### PR TITLE
Checkbox/Item Row Feature

### PR DESCRIPTION
**Pull request by m3ia**
- Added a basic checkbox (does not yet do anything)
- Changed EditItem.js to be ItemRow.js to make row a separate component from the table. Moved all buttons (Edit, Save, Delete) to ItemRow.js
- Fixed Enter-to-Save issue within ItemRow.js by changing KeyPress event to KeyDown
- Ensured page doesn't have to reload in order to see newly added items and edited items
- Edited items to handle overflowing text

Currently open Trello ticket: https://trello.com/c/mIrg39Zh

### CODE REVIEW

File: app/src/AddItemForm.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  };
+
+  // Submit sets setEditMode(false). User sees value with submitted input item.
+  const onSubmitClick = () => {
+    addNewItem(name);
+    setName("");
+    setEditMode(false);
+  };
+
+  // Want: When user clicks out => setEditMode(false) => editMode === false, a line with original value.
```
Comment (lemon-yellow) [inline app/src/AddItemForm.js:30]: What do you mean by "when user clicks out" ?

File: server/server.mjs [hunk 2]
```diff
@@ -9,22 +9,45 @@ const port = process.env.PORT || 4000;
 // items defines the routes
 const items = express.Router();
 
+// gets all items
 items.get("/", async (request, response) => {
   const items = await db.getItems();
   console.log("hello test test");
-  response.json(items);
+  response.status(201).json(items);
```
Comment (lemon-yellow) [inline server/server.mjs:16]: A successful Get request usually returns a status of `200`. I like to use this as a refresher https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses

File: server/server.mjs [hunk 3]
```diff
@@ -9,22 +9,45 @@ const port = process.env.PORT || 4000;
...
+});
+
+// gets one item
+items.get("/:id", async (request, response) => {
+  const { id } = request.params;
+  const item = await db.getItem(id);
+  // response.json(items.rows[0]);
+  response.status(201).json(item);
```
Comment (lemon-yellow) [inline server/server.mjs:24]: Ditto my earlier comment about the status code

**Review by lemon-yellow — APPROVED** [review]
Great work!

File: app/src/AddItemForm.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  return (
+    <>
+      {!editMode && (
+        <div
+          className="add-item-row"
+          onClick={() => {
+            onEditClick();
+          }}
+          onKeyPress={() => {
+            onEditClick();
```
Comment (lemon-yellow) [inline app/src/AddItemForm.js:61]: I noticed that if you remove this onKeyPress method, the "press Enter to save" functionality works. I think this keypress was being processed to setEdit to be true, when you wanted it to be false. Can you confirm that you still need this functionality?

File: app/src/AddItemForm.js (hunk 1 above)
Comment (m3ia) [reply app/src/AddItemForm.js:30]: Like when user clicks outside of the input box!

File: app/src/AddItemForm.js (hunk 4 above)
Comment (m3ia) [reply app/src/AddItemForm.js:61]: When I remove it, my linter tells me that if the div has an onClick() method, it also requires a key event for those who might need to tab through the page to interact with it in the case they cannot use a mouse! When I remove the line, the code won't compile for me.

**Comment by m3ia** [comment]
> Great work!

Thank you! 🙏🏽

File: app/src/AddItemForm.js (hunk 4 above)
Comment (lemon-yellow) [reply app/src/AddItemForm.js:61]: Makes sense! It's working for me now, which is great.

**Pull request merged**